packages/cli/src/lib/get-version.ts

```diff
@@ -0,0 +1,49 @@
+import { AbortError } from "@shopify/cli-kit/node/error";
+import { readFile } from "@shopify/cli-kit/node/fs";
+import { joinPath } from "@shopify/cli-kit/node/path";
+import { PackageJson } from "type-fest";
+import { getProjectPaths } from "./remix-config.js";
+
+interface Return {
+  currentVersion: string;
+  currentDependencies: Record<string, string>;
+}
+
+/**
+ * Gets the current @shopify/hydrogen version from the app's package.json
+ */
+export async function getHydrogenVersion({appPath}: {appPath: string}): Promise<Return> {
+  const {root} = getProjectPaths(appPath);
+  const packageJsonPath = joinPath(root, 'package.json');
+
+  let packageJson: PackageJson | undefined;
+
+  try {
+    packageJson = JSON.parse(await readFile(packageJsonPath));
+  } catch {
+    throw new AbortError(
+      'Could not find a valid package.json',
+      'Please make sure you are running the command in a npm project',
+    );
+  }
+
+  const currentDependenciesWithEmptyValues = {
+    ...packageJson?.dependencies,
+    ...packageJson?.devDependencies,
+  };
+
+  const currentDependencies = Object.fromEntries(
+    Object.entries(currentDependenciesWithEmptyValues).filter(([_, value]) => value)
+  ) as Record<string, string>;
```

Comment:
Had to add this type safety to ensure `Record<string, string>`. `currentDependenciesWithEmptyValues` can have undefined values, and `upgrade.ts` was having a hard time after.